### PR TITLE
Make iOS embedding to send full uri for deeplinks

### PR DESCRIPTION
**Pull request by chunhtai**
related https://github.com/flutter/flutter/issues/100624

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by vashworth — APPROVED** [review]
LGTM

File: shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm [hunk 1]
```diff
@@ -67,8 +67,9 @@ - (void)testLaunchUrl {
                             openURL:[NSURL URLWithString:@"http://myApp/custom/route?query=test"]
                             options:@{}];
   XCTAssertTrue(result);
-  OCMVerify([self.mockNavigationChannel invokeMethod:@"pushRouteInformation"
-                                           arguments:@{@"location" : @"/custom/route?query=test"}]);
+  OCMVerify([self.mockNavigationChannel
+      invokeMethod:@"pushRouteInformation"
+         arguments:@{@"location" : @"http://myApp/custom/route?query=test"}]);
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm:72]: Isn't it weird to be using the "http" protocol for this?

**Review by gaaclarke — COMMENTED** [review]
code looks good to me, just have a design question about using the http protocol (which was probably introduced in a framework pr?)

File: shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm (hunk 1 above)
Comment (chunhtai) [reply shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm:72]: An [universal link](https://developer.apple.com/documentation/xcode/allowing-apps-and-websites-to-link-to-your-content?language=objc) is either a http or https protocol.

Comment (chunhtai) [reply shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm:72]: This is the framework pr https://github.com/flutter/flutter/pull/119968, it just surface whatever url that is sent here to the developer in [RouteInformation.url](https://master-api.flutter.dev/flutter/widgets/RouteInformation/uri.html)

The goal is to let developer to be able to handle differently based on what uri to launch the app, for example they may want to block certain route if the app is launch from `example.com` but not when it is launch from `another-example.com`

**Comment by chinmaygarde** [comment]
Looks like @gaaclarke s question was answered. Can we land this please?

**Comment by chunhtai** [comment]
@astubenbord please use discord if you would like to discuss more. If you are using router API, `myapp://page/1` is stored in https://master-api.flutter.dev/flutter/widgets/RouteInformation/uri.html

**Comment by punxti** [comment]
Hi, 

My deeplink works fine on Android device. On iOS, it only opens the App, but not the routed page. I followed the configuration mentioned in https://docs.flutter.dev/cookbook/navigation/set-up-universal-links.

I found a similar issue mentioned in [post ](https://github.com/flutter/flutter/issues/100624), and this post seems to be resolving this matter. Do we need to do anything different with our code to incorporate the changes? 

Could you advise on this?

Many thanks

**Pull request merged**